scan/base.go

```diff
@@ -40,6 +42,163 @@ type base struct {
 	errs []error
 }
 
+func (l *base) scanWp() (err error) {
+	if len(l.ServerInfo.WpPath) == 0 && len(l.ServerInfo.WpToken) == 0 {
+		return nil
+	}
+
+	var unsecures []models.VulnInfo
+	if unsecures, err = detectWp(l.ServerInfo); err != nil {
+		l.log.Errorf("Failed to scan wordpress: %s", err)
+		return err
+	}
+	for _, i := range unsecures {
+		l.VulnInfos[i.CveID] = i
+	}
+
+	return err
+}
+
+//CveInfos hoge
+type CveInfos struct {
+	ReleaseDate     string    `json:"release_date"`
+	ChangelogURL    string    `json:"changelog_url"`
+	Status          string    `json:"status"`
+	Vulnerabilities []CveInfo `json:"vulnerabilities"`
+}
+
+//CveInfo hoge
+type CveInfo struct {
+	ID            int        `json:"id"`
+	Title         string     `json:"title"`
+	CreatedAt     string     `json:"created_at"`
+	UpdatedAt     string     `json:"updated_at"`
+	PublishedDate string     `json:"Published_date"`
+	VulnType      string     `json:"Vuln_type"`
+	References    References `json:"references"`
+	FixedIn       string     `json:"fixed_in"`
+}
+
+//References hoge
+type References struct {
+	URL []string `json:"url"`
+	Cve []string `json:"cve"`
+}
+
+func detectWp(c config.ServerInfo) (rs []models.VulnInfo, err error) {
+
+	var coreVuln []models.VulnInfo
+	if coreVuln, err = detectWpCore(c); err != nil {
+		return
+	}
+	for _, i := range coreVuln {
+		rs = append(rs, i)
+	}
+
+	var themeVuln []models.VulnInfo
+	if themeVuln, err = detectWpTheme(c); err != nil {
+		return
+	}
+	for _, i := range themeVuln {
+		rs = append(rs, i)
+	}
+
+	var pluginVuln []models.VulnInfo
+	if pluginVuln, err = detectWpPlugin(c); err != nil {
+		return
+	}
+	for _, i := range pluginVuln {
+		rs = append(rs, i)
+	}
+	return
+}
+
+func detectWpCore(c config.ServerInfo) (rs []models.VulnInfo, err error) {
+	cmd := fmt.Sprintf("wp core version --path=%s", c.WpPath)
+
+	var coreVersion string
+	if r := exec(c, cmd, noSudo); r.isSuccess() {
+		tmp := strings.Split(r.Stdout, ".")
+		coreVersion = strings.Join(tmp, "")
+		coreVersion = strings.TrimRight(coreVersion, "\r\n")
+		if len(coreVersion) == 0 {
+			return
+		}
+	}
+	cmd = fmt.Sprintf("curl -H 'Authorization: Token token=%s' https://wpvulndb.com/api/v3/wordpresses/%s", c.WpToken, coreVersion)
+	if r := exec(c, cmd, noSudo); r.isSuccess() {
+		data := map[string]CveInfos{}
+		if err = json.Unmarshal([]byte(r.Stdout), &data); err != nil {
+			return
+		}
+		for _, i := range data {
+			for _, e := range i.Vulnerabilities {
+				for _, k := range e.References.Cve {
+					k = "CVE-" + k
+				}
+
+				rs = append(rs, models.VulnInfo{
+					CveID: e.References.Cve[0],
+				})
+			}
+		}
+	}
+	return
+}
+
+type WpTheme struct {
```

Comment:
__[Golint]__ exported type WpTheme should have comment or be unexported

<sub>posted by Sider</sub>

Reply:
implement